x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/mapper/EvaluatorMapper.java

```diff
@@ -60,15 +67,37 @@ interface ToEvaluator {
      * in that it'll always call {@link Expression#fold}, but that's
      * good enough.
      */
-    default Object fold() {
-        return toJavaObject(toEvaluator(e -> driverContext -> new ExpressionEvaluator() {
+    default Object fold(Source source, FoldContext ctx) {
+        ToEvaluator toEvaluator = new ToEvaluator() {
             @Override
-            public Block eval(Page page) {
-                return fromArrayRow(driverContext.blockFactory(), e.fold())[0];
+            public ExpressionEvaluator.Factory apply(Expression expression) {
+                return driverContext -> new ExpressionEvaluator() {
+                    @Override
+                    public Block eval(Page page) {
+                        return fromArrayRow(driverContext.blockFactory(), expression.fold(ctx))[0];
+                    }
+
+                    @Override
+                    public void close() {}
+                };
             }
 
             @Override
-            public void close() {}
-        }).get(DriverContext.getLocalDriver()).eval(new Page(1)), 0);
+            public FoldContext foldCtx() {
+                return ctx;
+            }
+        };
+
+        DriverContext driverCtx = new DriverContext(
+            BigArrays.NON_RECYCLING_INSTANCE,
+            new BlockFactory(ctx.circuitBreakerView(source), BigArrays.NON_RECYCLING_INSTANCE)
```

Comment:
Those BigArrays aren't being used; are we sure that the blockfactory methods are enough? Some arrays in there are accounted by the BigArrays, right? So we're missing part of them.

Reply:
OOOH. Probably not. Let me have a look at wiring those in too.